deploy/app/legacyclaims.tf

```
@@ -0,0 +1,82 @@
+variable "legacy_claims_table_name" {
+  description = "The name of the DynamoDB table used by the legacy content-claims service"
+  type = string
+  default = "prod-content-claims-claims-v1"
+}
+
+variable "legacy_claims_bucket_name" {
+  description = "The name of the S3 bucket used by the legacy content-claims service"
+  type = string
+  default = "prod-content-claims-bucket-claimsv1bucketefd46802-1mqz6d8o7xw8"
+}
+
+variable "legacy_block_index_table_name" {
+  description = "The name of the legacy block index DynamoDB table"
+  type = string
+  default = "ep-v1-blocks-cars-position"
+}
+
+data "aws_s3_bucket" "legacy_claims_bucket" {
+  bucket = var.legacy_claims_bucket_name
+}
+
+data "aws_dynamodb_table" "legacy_claims_table" {
+  name = var.legacy_claims_table_name
+}
+
+data "aws_dynamodb_table" "legacy_block_index_table" {
+  name = "${terraform.workspace == "prod" ? "prod" : "staging"}-${var.legacy_block_index_table_name}"
+}
```

If you want to configure your own it'll always be given the prefix "staging". How about this instead?:

```suggestion
variable "legacy_block_index_table_name" {
  description = "The name of the legacy block index DynamoDB table"
  type = string
  default = "${terraform.workspace == "prod" ? "prod" : "staging"}-ep-v1-blocks-cars-position"
}

data "aws_s3_bucket" "legacy_claims_bucket" {
  bucket = var.legacy_claims_bucket_name
}

data "aws_dynamodb_table" "legacy_claims_table" {
  name = var.legacy_claims_table_name
}

data "aws_dynamodb_table" "legacy_block_index_table" {
  name = var.legacy_block_index_table_name
}
```

---

putting the conditional in the variable rather than the data source was the first thing I tried, but terraform complains about it.

In any case, using staging for anything not prod is intended. Since it's a resource not being set up within this configuration, it doesn't really make sense to make it dependent on the workspace used when deploying this service.

BTW, I'm failing to see how your suggestion behaves differently than the proposed change.

---

ah, I see what you mean now. Variables only allow literal values for their `default` attribute. I implemented a workaround using `locals`.